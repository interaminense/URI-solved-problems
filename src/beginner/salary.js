/**
 * Write a program that reads an employee's number, his/her worked hours number in a month and the
 * amount he received per hour. Print the employee's number and salary that he/she will receive at
 * end of the month, with two decimal places.
 *
 * Donâ€™t forget the space before and after the equal signal and after the U$.
 *
 * INPUT
 *
 * The input file contains 2 integer numbers and 1 value of floating point, representing the number,
 * worked hours amount and the amount the employee receives per worked hour.
 *
 * OUTPUT
 *
 * Print the number and the employee's salary, according to the given example, with a blank space
 * before and after the equal signal.
 */

const tmpl = (employers, salary) => `NUMBER = ${employers}\nSALARY = U$ ${salary}`;

const salary = (employers, hourNumber, valueByHour) => {
  const calcSalary = hourNumber * valueByHour;

  return tmpl(employers, calcSalary);
};

export { tmpl, salary };
