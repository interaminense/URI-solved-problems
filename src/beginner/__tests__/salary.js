import { salary, tmpl } from '../salary';

describe('Salary', () => {
  let employers = 0;
  let hourNumber = 0;
  let valueByHour = 0;

  it('Employers number is 25 and the salary is 550 ', () => {
    employers = 25;
    hourNumber = 100;
    valueByHour = 5.50;

    expect(salary(employers, hourNumber, valueByHour))
      .toEqual(tmpl(employers, 550));
  });

  it('Employers number is 1 and the salary is 4100 ', () => {
    employers = 1;
    hourNumber = 200;
    valueByHour = 20.50;

    expect(salary(employers, hourNumber, valueByHour))
      .toEqual(tmpl(employers, 4100));
  });

  it('Employers number is 6 and the salary is 2254.75 ', () => {
    employers = 6;
    hourNumber = 145;
    valueByHour = 15.55;

    expect(salary(employers, hourNumber, valueByHour))
      .toEqual(tmpl(employers, 2254.75));
  });
});
