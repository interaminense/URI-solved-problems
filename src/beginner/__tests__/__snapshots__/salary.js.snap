// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salary Employers number is 1 and the salary is 4100  1`] = `
"NUMBER = 1
SALARY = U$ 4100"
`;

exports[`Salary Employers number is 6 and the salary is 2254.75  1`] = `
"NUMBER = 6
SALARY = U$ 2254.75"
`;

exports[`Salary Employers number is 25 and the salary is 550  1`] = `
"NUMBER = 25
SALARY = U$ 550"
`;
